components/installer/pkg/steps/remove-kyma-components.go

```diff
@@ -2,55 +2,51 @@ package steps
 
 import (
 	"log"
+
+	"github.com/kyma-project/kyma/components/installer/pkg/config"
 )
 
 //RemoveKymaComponents .
-func (steps InstallationSteps) RemoveKymaComponents() {
+func (steps InstallationSteps) RemoveKymaComponents(installationData *config.InstallationData) {
 
 	log.Println("Removing Kyma resources...")
 
-	if err := steps.uninstallReleases(); err != nil {
+	if err := steps.uninstallReleases(installationData); err != nil {
 		log.Println("Error. Some releases may have not been removed.")
 		return
 	}
 
-	if err := steps.removeNamespaces(); err != nil {
-		log.Println("Error. Some namespaces may not have been removed")
-		return
-	}
-
 	log.Println("All Kyma resources have been successfully removed")
 }
 
-func (steps InstallationSteps) uninstallReleases() error {
+func (steps InstallationSteps) uninstallReleases(installationData *config.InstallationData) error {
 	log.Println("Uninstalling releases...")
-	releasesToBeDeleted := []string{"ec-default", "hmc-default", "dex", "core", "prometheus-operator", "istio", "cluster-essentials"}
+	installedReleases := make(map[string]bool)
 
-	for _, releaseName := range releasesToBeDeleted {
-		log.Println("Uninstalling release", releaseName)
-		uninstallReleaseResponse, uninstallError := steps.helmClient.DeleteRelease(releaseName)
-		if !steps.errorHandlers.CheckError("Uninstall Error: ", uninstallError) {
-			steps.helmClient.PrintRelease(uninstallReleaseResponse.Release)
-		}
+	releasesRes, err := steps.helmClient.ListReleases()
```

Comment:
Since it is for the user to decide which components will be uninstalled, there's no need to fetch Helm releases here.  IMHO the function should try to uninstall each component from the CR, logging some sort of a warning should there be no such release. Release exists = uninstall. Release not found = log warning and continue. This is how I see it.